isce2_topsapp/solid_earth_tides.py

```diff
@@ -6,96 +6,382 @@
+    orbit_xml : str | Path
+        Orbit xml file
 
-    acq_date = dates[0] if acq_type == 'reference' else dates[1]
+    Returns
+    -------
+    list[np.ndarray]
+      [t, pos_x, pos_y, pos_z, vel_x, vel_y, vel_z] each entry is a vector
+      denoting the state vector of the satellite at a given time, i.e. at time
+      t[k], the state vector is pos_x[k], ..., vel_z[k].
+    """
+    tree = ET.parse(orbit_xml)
+    root = tree.getroot()
+    data_block = root[1]
+    numOSV = len(data_block[0])
 
-    # get GUNW attributes
-    (wavelength,
-     z_meta,
-     inc_angle,
-     az_angle,
-     set_attrs) = get_gunw_attrs(gunw_path)
-
-    # compute differential SET ENU
-    # Use reference time twice (assuming exact same pass time)
-    tide_e, tide_n, tide_u = compute_enu_se_tide(acq_date,
-                                                 ref_time,
-                                                 set_attrs)
-
-    # compute SET LOS estimate for each height level
-    tide_los = np.zeros(inc_angle.shape, dtype=np.float32)
-    for i in range(len(z_meta)):
-        tide_los[i] = compute_los_solid_earth_tide(tide_e,
-                                                   tide_n,
-                                                   tide_u,
-                                                   inc_angle[i],
-                                                   az_angle[i],
-                                                   wavelength)
+    t = []
+    x = np.ones(numOSV)
+    y = np.ones(numOSV)
+    z = np.ones(numOSV)
+    vx = np.ones(numOSV)
+    vy = np.ones(numOSV)
+    vz = np.ones(numOSV)
 
-    solidtide_corr_ds = export_se_tides_to_dataset(gunw_path, tide_los)
+    for i, st in enumerate(data_block[0]):
+        t.append(
+            datetime.datetime.strptime(
+                st[1].text,
+                'UTC=%Y-%m-%dT%H:%M:%S.%f'
+            )
+        )
+
+        x[i] = float(st[4].text)
+        y[i] = float(st[5].text)
+        z[i] = float(st[6].text)
+        vx[i] = float(st[7].text)
+        vy[i] = float(st[8].text)
+        vz[i] = float(st[9].text)
+    t = np.array(t)
+    return [t, x, y, z, vx, vy, vz]
+
+
+def get_state_vector_arrays(orbit_xml: Union[str, Path]) -> list[StateVector]:
+    """Source: https://github.com/dbekaert/RAiDER/blob/dev/tools/RAiDER/losreader.py
+    """
+    state_vector_list = []
+    t, x, y, z, vx, vy, vz = read_ESA_Orbit_file(orbit_xml)
+    for idx in range(len(t)):
+        position = x[idx], y[idx], z[idx]
+        velocity = vx[idx], vy[idx], vz[idx]
+        state_vector = StateVector(time=t[idx],
+                                   position=position,
+                                   velocity=velocity)
+        state_vector_list.append(state_vector)
+
+    return state_vector_list
+
+
+def get_orbit_obj_from_orbit_xmls(orbit_xmls: list[Path],
+                                  slc_start_time: datetime.datetime,
+                                  pad_in_seconds: int = 600) -> Orbit:
```

Comment:
wonder how is the padding set in the isce topsAp, is it also 600 seconds?

Reply:
No idea...

Sim was able to track this down: https://github.com/dbekaert/RAiDER/blob/dev/tools/RAiDER/losreader.py#L188-L200

My rough understanding is that if there is not a padding then there are two solutions occurring at antipodal points on the globe.